src/index.js

```diff
@@ -150,6 +151,35 @@ class RedisStore {
       })
     })
   }
+
+  /**
+   * Get all entries in cache.
+   *
+   * @param {Function} fn
+   * @api public
+   */
+
+  getAll(fn = noop) {
+    const entries = []
+    const prefix = this.prefix
+
+    this.client.keys(`${prefix}*`, (err, keys) => {
```

Comment:
`keys` is not meant to be used in production. From the Redis documentation:

> This command is intended for debugging and special operations, such as changing your keyspace layout. Don't use KEYS in your regular application code. If you're looking for a way to find keys in a subset of your keyspace, consider using SCAN or sets.

Reply:
Asking out of best practices and curiosity:

We are currently using the `keys` functionality when [deleting all keys](https://github.com/cayasso/cacheman-redis/blob/master/src/index.js#L136). With the functionality of `keys` already being used, is there a reason why it's allowed in one place but not another?

As for what the docs recommend, here's the full explanation:
> Warning: consider KEYS as a command that should only be used in production environments with extreme care. It may ruin performance when it is executed against large databases. This command is intended for debugging and special operations, such as changing your keyspace layout. Don't use KEYS in your regular application code. If you're looking for a way to find keys in a subset of your keyspace, consider using SCAN or sets.

Since it's still an operation for redis itself, and we're just wrapping the redis api, why not expose this? It's ultimately up to the developer to utilize this functionality. I 100% see the downside to it, but should we block off features of a production type system, or leave that decision to the developer?

In regards to SCAN and SETS, the `SCAN` command will have to go through every key in the cache since it doesn't have a `filter` process of any sort. It seems like the operation time to get all keys in that instance would greatly increase since it could be multiple calls to get all the keys. `SETS` would probably use the `SMEMBERS` operation to grab all keys. From my testing locally, I could not get that to return all keys of the `prefix` we've defined in the source code. 

From these results, is there another way to get all keys from a redis cache if the proposed method isn't accepted?

Reply:
This accepted answer does a nice job of summarizing the differences between `SCAN` and `KEYS` - https://stackoverflow.com/questions/32603964/scan-vs-keys-performance-in-redis

As for the existing usage, that should be updated as well.

> I 100% see the downside to it, but should we block off features of a production type system, or leave that decision to the developer?

A very small subset of people are going to look at the source and see that it's using `keys` under the hood. An even smaller subset are going to know that `keys` should not be used in regular application code.